sdk/src/metrics/export/periodic_exporting_metric_reader.cc

```diff
@@ -89,19 +89,26 @@ void PeriodicExportingMetricReader::DoBackgroundWork()
 
 bool PeriodicExportingMetricReader::CollectAndExportOnce()
 {
-  std::atomic<bool> cancel_export_for_timeout{false};
-  auto future_receive = std::async(std::launch::async, [this, &cancel_export_for_timeout] {
-    Collect([this, &cancel_export_for_timeout](ResourceMetrics &metric_data) {
-      if (cancel_export_for_timeout)
-      {
-        OTEL_INTERNAL_LOG_ERROR(
-            "[Periodic Exporting Metric Reader] Collect took longer configured time: "
-            << export_timeout_millis_.count() << " ms, and timed out");
-        return false;
-      }
-      this->exporter_->Export(metric_data);
-      return true;
-    });
+  std::shared_ptr<std::atomic<bool>> cancel_export_for_timeout =
+      std::make_shared<std::atomic<bool>>(false);
+  auto keep_lifetime = shared_from_this();
```

Comment:
If I understand correctly, this code forces the use of a shared_ptr, and implicitly uses the shared_ptr internal reference counter to count how many exports are in flight.

Alternate solution:

* Implement a new member, `std::atomic<int> m_pending_exports`, to count how many exports are pending.
* When a new export starts, increment this counter
* When an export completes, decrement the counter
* In shutdown(), wait in a loop for the counter to reach 0

I think this will be better, with the following benefits:

* The code can check if exports are already pending before starting a new export, and raise an error, instead of piling up threads.
* The wait loop in shutdown can print warning messages with the current counter, so it is easier to debug, and understand that many exports are still in flight.
* This does not force `PeriodicExportingMetricReader` to become a shared_ptr